src/utils/SetupUtil.js

```diff
@@ -56,64 +56,56 @@ export default {
     let machineVersion = null;
     while (true) {
       try {
-        setupServerActions.started({started: false});
+        let ip = null;
```

Comment:
Can the linux setup be a separate function named `setupLinux`? Instead of interleaving it like this.